PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidNativeAd.java

```diff
@@ -0,0 +1,308 @@
+/*
+ *    Copyright 2020-2021 Prebid.org, Inc.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package org.prebid.mobile;
+
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.os.Handler;
+import android.os.Looper;
+import android.text.TextUtils;
+import android.util.Log;
+import android.view.View;
+import android.webkit.URLUtil;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class PrebidNativeAd {
+    private static final long NATIVE_AD_RESPONSE_EXPIRATION_TIME = 60 * 1000;
+    private String title;
+    private String description;
+    private String iconUrl;
+    private String imageUrl;
+    private String cta;
+    private String sponsoredBy;
+    private String clickUrl;
+    private ArrayList<String> imp_trackers;
+    private VisibilityDetector visibilityDetector;
+    private boolean expired;
+    private View registeredView;
+    private PrebidNativeAdEventListener listener;
+    private ArrayList<ImpressionTracker> impressionTrackers;
+
+
+    static PrebidNativeAd create(String cacheId) {
+        String content = CacheManager.get(cacheId);
+        if (!TextUtils.isEmpty(content)) {
+            try {
+                JSONObject details = new JSONObject(content);
+                String admStr = details.getString("adm");
+                JSONObject adm = new JSONObject(admStr);
+                JSONArray asset = adm.getJSONArray("assets");
+                final PrebidNativeAd ad = new PrebidNativeAd();
+                CacheManager.registerCacheExpiryListener(cacheId, new CacheManager.CacheExpiryListener() {
+                    @Override
+                    public void onCacheExpired() {
```

Comment:
This logic does not look correct.
If a publisher is displaying the ad on the screen after registration, this would cause issues.

Reply:
Done.